docs/source/_comparison_generator.py

```diff
@@ -8,8 +8,9 @@
 
 
 def _get_functions(obj, exclude=None):
+    children = obj.__all__ if hasattr(obj, '__all__') else dir(obj)
```

Comment:
`__all__` can contain "private" names that start with `_`.  How about moving the condition `not n.startswith('_')` here? (For example, which I admit not really a good example, `'__version__' in numpy.__all__`).

I'm not sure if default `__all__` contains submodule names, but it looks always fine to exclude submodules from the comparison table.

```suggestion
    try:
        children = obj.__all__
    except AttributeError:
        children = [n for n in dir(obj) if not n.startswith('_')]
```